core/src/avm2/object/bytearray_object.rs

```diff
@@ -14,14 +15,34 @@ pub fn byte_array_allocator<'gc>(
     class: ClassObject<'gc>,
     activation: &mut Activation<'_, 'gc>,
 ) -> Result<Object<'gc>, Error<'gc>> {
+    let storage = if let Some((movie, id)) = activation
+        .context
+        .library
+        .avm2_class_registry()
+        .class_symbol(class)
+    {
+        if let Some(lib) = activation.context.library.library_for_movie(movie) {
+            if let Some(Character::BinaryData(binary_data)) = lib.character_by_id(id) {
+                Some(ByteArrayStorage::from_vec(binary_data.as_ref().to_vec()))
+            } else {
+                None
```

Comment:
This else is still reachable: for example, a class that extends `ByteArray` but doesn't correspond to a `BinaryData` in the symbol class.

Reply:
This only occurs if we're successfully looked up the `class_symbol`.